File: javascript/challenges/231-the-time-is-always-right-to-do-what-is-right-dr/index.test.ts
import { expect, describe, it } from "vitest";
import { wordleGuess } from ".";

// // Talk is cheap, show me the code
describe.concurrent("Check errors", () => {
  it("test when input has less than 5 characters", async () => {
    expect(() => wordleGuess("fake", "angel")).toThrowError(
      "Each guess must be a valid 5-letter word."
    );
  });

  it("test when solution has less than 5 characters", async () => {
    expect(() => wordleGuess("angel", "fake")).toThrowError(
      "Each guess must be a valid 5-letter word."
    );
  });
});

describe.concurrent("Check solutions with word 'fudge'", () => {
  const solutionWord = "fudge";

  it("'reads' must has two yellow coincidences", async () => {
    expect(wordleGuess("reads", solutionWord)).toBe("⬛🟨⬛🟨⬛");
  });

  it("'deeds' must has two yellow coincidences", async () => {
    expect(wordleGuess("deeds", solutionWord)).toBe("🟨🟨⬛⬛⬛");
  });

  it("'lodge' must has three green coincidences", async () => {
    expect(wordleGuess("lodge", solutionWord)).toBe("⬛⬛🟩🟩🟩");
  });

  it("test 100% coincidence", async () => {
    expect(wordleGuess("fudge", solutionWord)).toBe("🟩🟩🟩🟩🟩");
  });
});

describe.concurrent("Check solutions with word 'perps'", () => {
  const solutionWord = "perps";

  it("'poppy' must has two green coincidences", async () => {
    expect(wordleGuess("poppy", solutionWord)).toBe("🟩⬛⬛🟩⬛");
  });

  it("'eaeee' must has one yellow coincidence", async () => {
    expect(wordleGuess("eaeee", solutionWord)).toBe("🟨⬛⬛⬛⬛");
  });

  it("'erpsp' must has five yellow coincidences", async () => {
    expect(wordleGuess("erpsp", solutionWord)).toBe("🟨🟨🟨🟨🟨");
  });
});

File: ruby/challenges/231-the-time-is-always-right-to-do-what-is-right-dr/index.test.rb
# frozen_string_literal: true

require 'minitest/autorun'
require_relative 'index'

# Talk is cheap, show me the code
class WordleTest < Minitest::Test
  def test_raises_error_if_invalid_solution
    assert_raises(StandardError) { Wordle.new 'fake' }
  end

  def test_raises_error_if_invalid_guess
    assert_raises(StandardError) do
      Wordle.new 'angel'
      wordle.guess('fake')
    end
  end

  def test_word_reads_with_solution_fudge
    wordle = Wordle.new 'fudge'
    assert_equal '⬛🟨⬛🟨⬛', wordle.guess('reads')
  end

  def test_word_deeds_with_solution_fudge
    wordle = Wordle.new 'fudge'
    assert_equal '🟨🟨⬛⬛⬛', wordle.guess('deeds')
  end

  def test_word_lodge_with_solution_fudge
    wordle = Wordle.new 'fudge'
    assert_equal '⬛⬛🟩🟩🟩', wordle.guess('lodge')
  end

  def test_full_coincidence
    wordle = Wordle.new 'fudge'
    assert_equal '🟩🟩🟩🟩🟩', wordle.guess('fudge')
  end

  def test_word_poppy_with_solution_perps
    wordle = Wordle.new 'perps'
    assert_equal '🟩⬛⬛🟩⬛', wordle.guess('poppy')
  end

  def test_word_eaeee_with_solution_perps
    wordle = Wordle.new 'perps'
    assert_equal '🟨⬛⬛⬛⬛', wordle.guess('eaeee')
  end

  def test_word_erpsp_with_solution_perps
    wordle = Wordle.new 'perps'
    assert_equal '🟨🟨🟨🟨🟨', wordle.guess('erpsp')
  end
end
